core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -1754,9 +1754,14 @@ public void removeNode(@Nonnull Node n) throws IOException {
     /**
      * Updates an existing {@link Node} on disk.
      * @see Nodes#updateNode
+     * @since 1.634
      */
     public boolean updateNode(Node n) throws IOException {
-        return nodes.updateNode(n);
+        if ( nodes.updateNode(n) ){
```

Comment:
@aheritier I do not see a circular reference or any issues with it moving to `Nodes` (as I said `Nodes` is already doing this for the case you update a node via the web UI)